### PR TITLE
WT-8848, WT-10027, WT-9774 and WT-9879 v6.0 Backport 

### PR DESCRIPTION
**Pull request by ruby-oujo**
WT-8848 Add session statistics that counts a transaction's dirty bytes. 
(cherry picked from commit 3112dccffa27bb8152bbc8ed22f63a0361cf28d7)

[WT-10027](https://jira.mongodb.org/browse/WT-10027) Fix session txn statistic to exclude eviction work.
(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/d5edce423f92ef3208b711735b803cc712ff807b )

[WT-9879](https://jira.mongodb.org/browse/WT-9879) Fix overcounting of session txn dirty bytes statistic
(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/20386a28e2b57b1f8d2f4c3d8b894180876911db )

[WT-9774](https://jira.mongodb.org/browse/WT-9774) Avoid rolling back transactions in test_stat08 
(Manual merge from commit [`bb5c03d`](https://github.com/wiredtiger/wiredtiger/commit/bb5c03d59037747b3c477bb58e4c82eb961e5589))

### CODE REVIEW

**Comment by lukech** [comment]
@ruby-oujo should this ticket be backported together with the follow-on changes (WT-9879 and WT-10027)?

**Comment by ruby-oujo** [comment]
@lukech that is a good idea, I will do that now.

**Comment by ruby-oujo** [comment]
The changes of WT-9774 ([BACKPORT-14293](https://jira.mongodb.org/browse/BACKPORT-14293)) also need to be backported alongside this group of changes as it alters `test_stat08.py` before WT-9879 so backporting WT-9774 now to avoid overwriting changes and the changes are required otherwise the current changes fail `unit-test-bucket-06`.

File: src/include/txn.h [hunk 1]
```diff
@@ -19,8 +19,9 @@
  * the session get rollback reason API call. Users of the API could have a dependency on the format
  * of these messages so changing them must be done with care.
  */
-#define WT_TXN_ROLLBACK_REASON_CACHE "oldest pinned transaction ID rolled back for eviction"
 #define WT_TXN_ROLLBACK_REASON_CONFLICT "conflict between concurrent operations"
+#define WT_TXN_ROLLBACK_REASON_OLDEST_FOR_EVICTION \
```
Comment (sulabhM) [inline src/include/txn.h:23]: I don't think we should rename this macro. This was done as part of another change: https://github.com/wiredtiger/wiredtiger/commit/de7db74e12cdedd6d026e7674c466d95e47024bd

We should keep the same name as before, and hence no reordering. This naming also matches the corresponding entry in MongoDB code:
https://github.com/mongodb/mongo/blob/v6.0/src/mongo/db/storage/wiredtiger/wiredtiger_util.cpp#L64

Let's keep the name.

File: src/txn/txn.c [hunk 2]
```diff
@@ -2367,7 +2367,7 @@ __wt_txn_is_blocking(WT_SESSION_IMPL *session)
      * Check if either the transaction's ID or its pinned ID is equal to the oldest transaction ID.
      */
     return (txn_shared->id == global_oldest || txn_shared->pinned_id == global_oldest ?
-        __wt_txn_rollback_required(session, WT_TXN_ROLLBACK_REASON_CACHE) :
+        __wt_txn_rollback_required(session, WT_TXN_ROLLBACK_REASON_OLDEST_FOR_EVICTION) :
```
Comment (sulabhM) [inline src/txn/txn.c:2370]: As commented above, let's keep the former naming.

**Review by sulabhM — CHANGES_REQUESTED** [review]

File: src/txn/txn.c (hunk 2 above)
Comment (ruby-oujo) [reply src/txn/txn.c]: addressed here [f789e91](https://github.com/wiredtiger/wiredtiger/pull/8665/commits/f789e9155171960af9babea55760494256cc3ff7)

File: src/include/txn.h (hunk 1 above)
Comment (ruby-oujo) [reply src/include/txn.h]: addressed [f789e91](https://github.com/wiredtiger/wiredtiger/pull/8665/commits/f789e9155171960af9babea55760494256cc3ff7)

**Review by sulabhM — APPROVED** [review]

**Pull request merged**